automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/security/AuthDomainSpec.scala

```diff
@@ -725,15 +734,19 @@ class AuthDomainSpec extends FreeSpec /*with ParallelTestExecution*/ with Matche
 
       withBillingProject("auth-domain-spec") { projectName =>
         withGroup("AuthDomain") { groupName =>
-          withWorkspace(projectName, "AuthDomainSpec_reject", Set(groupName)) { workspaceName =>
-            api.groups.addUserToGroup(groupName, user.email, GroupRole.Member)
-            checkNoAccess(user, projectName, workspaceName)
+          withCleanUp {
+            withWorkspace(projectName, "AuthDomainSpec_reject", Set(groupName)) { workspaceName =>
+              api.groups.addUserToGroup(groupName, user.email, GroupRole.Member)
+              register cleanUp api.groups.removeUserFromGroup(groupName, user.email, GroupRole.Member)
 
-            api.billing.addUserToBillingProject(projectName, user.email, BillingProjectRole.Owner)
-            checkVisibleAndAccessible(user, projectName, workspaceName)
+              checkNoAccess(user, projectName, workspaceName)
 
-            api.billing.removeUserFromBillingProject(projectName, user.email, BillingProjectRole.Owner)
-            checkNoAccess(user, projectName, workspaceName)
+              api.billing.addUserToBillingProject(projectName, user.email, BillingProjectRole.Owner)
```

Comment:
A little unrelated but this maybe should also have:
```
register cleanUp api.billing.removeUserFromBillingProject(projectName, user.email, BillingProjectRole.Owner)
```

Reply:
Replacing or in addition to
```
api.billing.removeUserFromBillingProject(projectName, user.email, BillingProjectRole.Owner)
```
on line 747?

Reply:
Assuming replacing based on the pattern for the other one

Reply:
Actually it's in addition to because of the `checkNoAccess` at the end. I would expect that assertion to fail in the test's current form.

Reply:
Gotcha, fell into mindless pattern recognition